### PR TITLE
Transfering bodies while being logged out

### PR DESCRIPTION
**Pull request by Aranclanos**
WIP transfering bodies while being logged out

### CODE REVIEW

File: UnityProject/Assets/Scripts/Managers/NetworkManagement/SpawnHandler.cs [hunk 1]
```diff
@@ -12,60 +12,69 @@ public static void SpawnViewer(NetworkConnection conn, short playerControllerId,
...
-		NetworkServer.ReplacePlayerForConnection(conn, ghost, playerControllerId);
-		TriggerEventMessage.Send(ghost, EVENT.GhostSpawned);
-		if (connectedPlayer.Script.PlayerSync != null) {
-			connectedPlayer.Script.PlayerSync.NotifyPlayers(true);
+		if (connectedPlayer == ConnectedPlayer.Invalid)
+		{
+			//conn = new NetworkConnection();
+			ConnectedPlayer loggedOffPlayer = PlayerList.Instance.UpdateLoggedOffPlayer(newBody, oldBody);
+			conn = loggedOffPlayer.Connection;
+			NetworkServer.ReplacePlayerForConnection(conn, newBody, 0); // <- bajillion warnings
```
Comment (Aranclanos) [inline UnityProject/Assets/Scripts/Managers/NetworkManagement/SpawnHandler.cs]: These three lines are the part where I can't wrap my head around, the purpose is to leave a clientless body. Both dummy creation and body transfer while logged out end up here. The new body was already created but if I dont call for `NetworkServer.ReplacePlayerForConnection()`, the new mob just starts runtiming to hell and back. If I DO call for this function, unity gets 1k warnings that even lag the editor for half a second.
Need help!

Comment (DooblyNoobly) [reply UnityProject/Assets/Scripts/Managers/NetworkManagement/SpawnHandler.cs]: What are the warnings?

Comment (Aranclanos) [reply UnityProject/Assets/Scripts/Managers/NetworkManagement/SpawnHandler.cs]: image
its a spam of these bad boys, all the same

Comment (DooblyNoobly) [reply UnityProject/Assets/Scripts/Managers/NetworkManagement/SpawnHandler.cs]: So its all coming from here: https://github.com/unitystation/unitystation/blob/2f53e62956dbfa191cb6b3be42b0bff2c2084962/UnityProject/Assets/Scripts/Messages/Server/ServerMessage.cs#L24

Interesting....

Comment (DooblyNoobly) [reply UnityProject/Assets/Scripts/Managers/NetworkManagement/SpawnHandler.cs]: See if anything in this thread helps: https://forum.unity.com/threads/reconnect-client-to-server.436033/

**Comment by Aranclanos** [comment]
It's finished, the white bodies character preferences is another issue that it'll fix in another PR

**Comment by fomalsd** [comment]
what cases does this pr address? 
like, someone getting cloned while being offline? any other scenarios?

**Comment by Aranclanos** [comment]
as for now dying and respawning while being logged off, also makes the F7 husk spawning not pop 1k warnings

**Review by fomalsd — COMMENTED** [review]
Code generally looks ok, except maybe for a few missed opportunities of reducing GetComponent<> calls (as PlayerScript already has references for RegisterPlayer, ObjectBehaviour etc)
But I found out that spawning multiple husks doesn't add more players to PlayerList (only the first one is added), that doesn't seem like a good sign

**Comment by fomalsd** [comment]
ok, let's test it on steam server

**Pull request merged**